integration-tests/suites/check_duplicate_endpoints.go

```diff
@@ -0,0 +1,111 @@
+package suites
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/stackrox/collector/integration-tests/suites/common"
+	"github.com/stretchr/testify/assert"
+)
+
+type CheckDuplicateEndpointsTestSuite struct {
+	IntegrationTestSuiteBase
+	serverContainer string
+}
+
+func (s *CheckDuplicateEndpointsTestSuite) waitForEndpoints() {
+	_, err := s.GetEndpoints(s.serverContainer)
+	count := 0
+	maxCount := 60
+
+	for err != nil {
+		time.Sleep(1 * time.Second)
+		_, err = s.GetEndpoints(s.serverContainer)
+		count += 1
+		if count == maxCount {
+			break
+		}
+	}
+}
+
+func (s *CheckDuplicateEndpointsTestSuite) killSocatProcess(port int) {
+	output, err := s.execContainer("socat", []string{"/bin/sh", "-c", "ps | grep socat.*LISTEN:" + strconv.Itoa(port) + ",fork | head -1 | awk '{print $1}'"})
+	outputReplaced := strings.Replace(output, "\"", "", -1)
+	outputTrimmed := strings.Trim(outputReplaced, " ")
+	pid := strings.Split(outputTrimmed, " ")[0]
+	s.Require().NoError(err)
+	_, err = s.execContainer("socat", []string{"/bin/sh", "-c", "kill -9 " + pid})
+	s.Require().NoError(err)
+}
+
+// The desired time line of events in this test is the following. Start a process that opens a port.
+// When this endpoint is reported in the mock grpc server we know that a scrape has occured. This is
+// time 0. We then start another process that opens a different port, 81 slightly after t=0.
+// The scrape interval is 8 seconds so at t=8 the endpoint with port 81 should be reported. At t=10
+// seconds the process is killed. At t=10 seconds an identical command that opened port 81 is
+// executed. At t=16 at the second scrape interval nothing should be reported, since the port was
+// opened and closed in the same interval. We expect there to be two reported endpoints.
+func (s *CheckDuplicateEndpointsTestSuite) SetupSuite() {
+
+	s.metrics = map[string]float64{}
+	s.executor = common.NewExecutor()
+	s.StartContainerStats()
+	s.collector = common.NewCollectorManager(s.executor, s.T().Name())
+
+	s.collector.Env["COLLECTOR_CONFIG"] = `{"logLevel":"debug","turnOffScrape":false,"scrapeInterval":8}`
+	s.collector.Env["ROX_PROCESSES_LISTENING_ON_PORT"] = "true"
+
+	err := s.collector.Setup()
+	s.Require().NoError(err)
+
+	err = s.collector.Launch()
+	s.Require().NoError(err)
+	time.Sleep(30 * time.Second)
+
+	processImage := common.QaImage("quay.io/rhacs-eng/qa", "socat")
+
+	containerID, err := s.launchContainer("socat", processImage, "/bin/sh", "-c", "socat TCP-LISTEN:80,fork STDOUT")
+
+	s.Require().NoError(err)
+	s.serverContainer = common.ContainerShortID(containerID)
+	s.waitForEndpoints()
+
+	_, err = s.execContainer("socat", []string{"/bin/sh", "-c", "socat TCP-LISTEN:81,fork STDOUT &"})
+	s.Require().NoError(err)
+	time.Sleep(10 * time.Second)
+	s.killSocatProcess(81)
+	_, err = s.execContainer("socat", []string{"/bin/sh", "-c", "socat TCP-LISTEN:81,fork STDOUT &"})
```

Comment:
Use a separate string so to reduce duplication
```suggestion
	command := []string{"/bin/sh", "-c", "socat TCP-LISTEN:81,fork STDOUT &"}
	
	_, err = s.execContainer("socat", command)
	s.Require().NoError(err)
	time.Sleep(10 * time.Second)
	s.killSocatProcess(81)
	
	_, err = s.execContainer("socat", command)
```

Reply:
Done